src/dotnet/CarbonAware.WebApi/Filters/HttpResponseExceptionFilter.cs

```
@@ -11,6 +11,12 @@ public class HttpResponseExceptionFilter : IExceptionFilter
     private ILogger<HttpResponseExceptionFilter> _logger;
     private ActivitySource _activitySource;
 
+    private static Dictionary<string, int> EXCEPTION_STATUS_CODE_MAP = new Dictionary<string, int>()
+    {
+        { "ArgumentException", (int)HttpStatusCode.BadRequest },
+        { "NotImplementedException", (int)HttpStatusCode.NotImplemented },
+    };
+
     public HttpResponseExceptionFilter(ILogger<HttpResponseExceptionFilter> logger, ActivitySource activitySource)
```

Not sure if all the cases are catch here, but I noticed that 'system' exceptions are been swallowed by this filter.  (i.e NullPointerExceptions, ArrayOutOfBoundExceptions, ...) Something maybe to look for on a bug instead.

---

That's a good point.  From a user standpoint we do want to swallow these to ensure everything is a structured response but we should walk through all of the implications of this.

As a starting point I've added `activity?.SetStatus(ActivityStatusCode.Error, context.Exception.Message);` to those catchall cases to ensure they are properly flagged in telemetry, but we may want to also consider not exposing any details around an unhandled exception in production and being more verbose in development for debugging purposes.  I will file something to capture this.

---

Issue 359 captures this now